x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/planner/QueryTranslatorTests.java

```diff
@@ -1334,6 +1334,7 @@ public void testTranslateStDistanceToQuery() {
         assertEquals(20.0, gq.lat(), 0.00001);
         assertEquals(10.0, gq.lon(), 0.00001);
         assertEquals(25.0, gq.distance(), 0.00001);
+        optimizeAndPlan(p);
```

Comment:
? - there are no extra assertions after this call.

Reply:
The `optimizeAndPlan(p)` was introduced as part of the https://github.com/elastic/elasticsearch/commit/9fc11a555fad3c636edff71b5a1b84d0cdc5e1c0 commit to make sure we exercise the `SqlTranslationHandler` (the `translate()` does not do that) and check if it will produce a `RuntimeException`. Added an extra check on the generated query.

Added a test for the Verifier to capture the current behaviour: the `WHERE ST_Distance(nested.point, ...) < X` is not even allowed today (`WHERE isn't (yet) compatible with scalar functions on nested fields`).